sources/packages/backend/libs/services/src/clamav/services/clamav.service.ts

```diff
@@ -22,27 +27,25 @@ export class ClamAVService {
   /**
    * Scans the file for viruses.
    * @param stream stream to be scanned.
-   * @param processSummary process summary logs.
    * @returns boolean true if the stream is virus infected,
    * false if the stream is not virus infected, and
    * null if the stream could not be scanned for viruses.
    */
-  async scanFile(
-    stream: Readable,
-    processSummary: ProcessSummary,
-  ): Promise<boolean> {
+  async scanFile(stream: Readable): Promise<VirusScanCode> {
     try {
       if (!this.scanner) {
         await this.initClam();
       }
       const { isInfected } = await this.scanner.scanStream(stream);
-      return isInfected;
+      return { isInfected };
     } catch (err) {
       if (err.code === "ECONNREFUSED") {
-        processSummary.error("Connection to ClamAV server failed.");
-        return;
+        return { errorCode: CONNECTION_FAILED };
       }
-      processSummary.error(err);
+      if (err.code === "ENOTFOUND") {
```

Comment:
What is the intention or benefit of this `catch` block? It is just renaming the lib error codes?

Reply:
I made the change, but E2E test became unavailable because it mocks the method
![image](https://github.com/user-attachments/assets/a551c879-1684-4ca6-9eed-890f2d3860aa)

Reply:
It is no longer a problem since the code was refactored hence closing it.